src/dataflow-types/src/types.rs

```diff
@@ -79,7 +79,16 @@ pub struct BuildDesc<View> {
 #[derive(Clone, Debug, Default, Serialize, Deserialize)]
 pub struct DataflowDescription<View> {
     /// Sources made available to the dataflow.
-    pub source_imports: BTreeMap<GlobalId, crate::types::sources::SourceDesc>,
+    ///
+    /// Each identifier is associated with a source description, and an optional
+    /// bundle of linear operators to apply to the source once instantiated.
+    pub source_imports: BTreeMap<
+        GlobalId,
+        (
+            crate::types::sources::SourceDesc,
+            Option<crate::types::LinearOperator>,
+        ),
+    >,
```

Comment:
Thoughts on ultimately coming up with a name for this tuple? 

I find stuff like

```rust
self.source_imports.insert(id, (description, None));
```

a little hard to read, vs e.g.

```rust
self.source_imports.insert(id, SourceImportDesc {
    desc,
    linear_operator: None,
})
```

100% no need to do anything now, just wanted to see if you had some moral objection to the idea down the road.

Reply:
Yeah, could certainly do `SourceInstantiation` or something like that.

Reply:
Oh wait, you wrote the same thing basically. Idk why I got to responding after reading the first sentence, sorry.